applications/jonny5/src/j5_balcraw_worker.erl

```diff
@@ -0,0 +1,118 @@
+    end.
+
+-spec crawler(api_binaries()) -> no_return().
+crawler([]) -> exit("work done");
+crawler([Account|Accounts]) ->
+    maybe_disconnect_account(Account),
+    Delay = whapps_config:get_integer(?APP_NAME, <<"balance_crawler_interaccount_delay_ms">>, ?DEFAULT_INTERACCOUNT_DELAY),
+    timer:sleep(Delay),
+    crawler(Accounts).
+
+-spec maybe_disconnect_account(ne_binary()) -> 'ok'.
+maybe_disconnect_account(AccountId) ->
+    Limits = j5_limits:get(AccountId),
+    DisconnectActiveCalls = j5_limits:disconnect_active_calls(Limits),
+    case wh_util:is_true(DisconnectActiveCalls) of
+        'false' -> 'ok';
+        'true' ->
+            case j5_per_minute:maybe_credit_available(0, Limits, 'true') of
+                'true' -> 'ok';
+                'false' -> disconnect_account(AccountId)
+            end
+    end.
+
+-spec disconnect_account(ne_binary()) -> 'ok'.
+disconnect_account(AccountId) ->
+    case j5_channels:per_minute(AccountId) of
+        Count when Count > 0 ->
+            lager:debug("account ~p has ~p per-minute calls, disconnect them",[AccountId, Count]),
+            maybe_disconnect_channels(j5_channels:account(AccountId));
+        _ ->
+            lager:debug("account ~p doesn't have any per-minute call",[AccountId]),
+            'ok'
+    end.
+
+-spec maybe_disconnect_channels(j5_channels:channels()) -> 'ok'.
+maybe_disconnect_channels([]) -> 'ok';
+maybe_disconnect_channels([Channel|Channels]) ->
+    Props = j5_channels:to_props(Channel),
+    case props:get_binary_value(<<"Account-Billing">>, Props) =:= <<"per_minute">>
+         orelse props:get_binary_value(<<"Reseller-Billing">>, Props) =:= <<"per_minute">>
+    of
+        'false' -> maybe_disconnect_channels(Channels);
+        'true' ->
+            disconnect_channel(Props),
+            maybe_disconnect_channels(Channels)
+    end.
+
+-spec disconnect_channel(wh_proplist()) -> 'ok'.
+disconnect_channel(Props) ->
+    CallId =  props:get_ne_binary_value(<<"Call-ID">>, Props),
+    HangupDelay = case whapps_config:get_is_true(?APP_NAME, <<"balance_crawler_delayed_hangup">>, true) of
+                      'true' -> get_hangup_delay(Props);
+                      'false' -> 0
+                  end,
+    lager:debug("call id ~p, account billing ~p, reseller billing ~p",[CallId, props:get_binary_value(<<"Account-Billing">>, Props), props:get_binary_value(<<"Reseller-Billing">>, Props)]),
+    try_disconnect_call(CallId, HangupDelay).
+
+-spec get_hangup_delay(wh_proplist()) -> integer().
+get_hangup_delay(Props) ->
+    case props:get_integer_value(<<"Answered-Timestamp">>, Props) of
+        'undefined' -> 0;
+        AnsweredTimestamp ->
+            CallTime = wh_util:current_tstamp() - AnsweredTimestamp,
+            RateMinimum = props:get_integer_value(<<"Rate-Minimum">>, Props),
+            RateIncrement = props:get_integer_value(<<"Rate-Increment">>, Props),
+            MaxCallTime = ((CallTime div RateIncrement) + 1) * RateIncrement,
```

Comment:
The idea is round MaxCallTime to next RateIncrement, exactly as it will be billed.
Example:
current call time = 141 seconds
RateIncrement = 60
If we disconnect immediately then call will be billed as 180 seconds.
((CallTime div RateIncrement) + 1) * RateIncrement = 180 (we delay hangup for 39 seconds).

CallTime + RateIncrement = 201 seconds (delay hangup for 60 seconds), and it will be billed as 240 seconds.